### PR TITLE
Progress with UnboundBL!

### PR DESCRIPTION
**Pull request by alectrocute**
Still got some testing to do. Any help counts, I really appreciate the community's help so far getting this plugin up and running.

### CODE REVIEW

**Comment by mimugmail** [comment]
Why do you remove the copyright, there should be at least one ..

**Comment by alectrocute** [comment]
@mimugmail I forget why I removed that; I think I cleared it off so I could see the source in a less messy manner but then forgot to paste it back in. Oops!

**Comment by mimugmail** [comment]
The file looks like this .. is this intended?

```
server:
local-data: "0-act.channel.facebook.com A 0.0.0.0"
server:
local-data: "0-edge-chat.facebook.com A 0.0.0.0"
server:
local-data: "0.0.0.0 A 0.0.0.0"
server:
local-data: "0.r.msn.com A 0.0.0.0"
server:
local-data: "0.twitter.com A 0.0.0.0"
server:
local-data: "000.0x1f4b0.com A 0.0.0.0"
server:
local-data: "000.gaysexe.free.fr A 0.0.0.0"
server:
local-data: "000free.us A 0.0.0.0"
server:
local-data: "000owamail0.000webhostapp.com A 0.0.0.0"
server:
local-data: "000tristanprod.free.fr A 0.0.0.0"
server:
local-data: "001.0x1f4b0.com A 0.0.0.0"
server:
local-data: "002.0x1f4b0.com A 0.0.0.0"
server:
local-data: "003.0x1f4b0.com A 0.0.0.0"
server:
local-data: "004.0x1f4b0.com A 0.0.0.0"

```

Also the spinner of Save button is outside of the button :)

File: dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [refresh]
-command:/usr/local/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh
+command:sh /usr/local/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh
```
Comment (fabianfrz) [inline dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf]: this is not needed, you probably have not set up the executable bit in the file permission correctly.

**Comment by beren12** [comment]
Would love to see this finished, thanks for the code!

**Comment by mimugmail** [comment]
@alectrocute any news? :)

**Comment by alectrocute** [comment]
@mimugmail Working on it! :)

**Comment by alectrocute** [comment]
@mimugmail @fabianfrz 

Big progress, it should be totally functional now and working well.

Only thing I'm worried about is having the generated .sh files executable (chmod +x in the Makefile???). If that's not a problem, then disregard.

So excited!

**Comment by alectrocute** [comment]
Is there any way to make the tokenized input field wider and consist of more rows without mucking about in the CSS (eg. wrapping it in a div)?

I’d like to add features in the very near future:

- Block mode (0.0.0.0, reject, pixelserv & nginx).
- Unbound log parser which shows most recently / frequently blocked domains and which hostname attempted to access.
- Cron settings to automatically pull blocklists.
- Turn off/on at different times, basic scheduling.
- A list of trusted lists that can easily be added to the blacklist field without having to do googling and research.

But right now, I feel that pending your review(s), we have a minimum viable product that will be very beneficial for OPNsense users.

File: dns/unboundbl/src/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
+# init counter for debugging purposes
+cnt=0
+
+echo "Removing old files..."
+[ -f /var/unbound/dnsbl.conf ] && rm -f /var/unbound/dnsbl.conf
+[ -f /tmp/hosts.working ] && rm -f /tmp/hosts.working
 
 # include config-generated blacklist/whitelist, commas replaced with spaces
-source /usr/local/opnsense/scripts/OPNsense/Unboundbl/data.sh
+. /usr/local/opnsense/scripts/OPNsense/Unboundbl/data.sh
```
Comment (beren12) [inline dns/unboundbl/src/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh]: I'm on the fence with this, "source" is more noticeable than "." but I use "." in my own stuff pretty often

File: dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [refresh]
-command:sh /usr/local/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh
+command:/usr/local/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh
```
Comment (beren12) [inline dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf]: Nothing wrong with calling the script directly from sh, it's one 3 chars longer

Comment (mimugmail) [reply dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf]: Franco will set the X bits when this is in master, no worries :)

File: dns/unboundbl/src/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh (hunk 2 above)
Comment (alectrocute) [reply dns/unboundbl/src/opnsense/scripts/OPNsense/Unboundbl/UnboundBL.sh]: I ran into some strange errors while importing variables via “.”, while “source” never complained and acted predictably.

File: dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf (hunk 3 above)
Comment (alectrocute) [reply dns/unboundbl/src/opnsense/service/conf/actions.d/actions_unboundbl.conf]: Lol this is what I thought, but I’m trusting @mimugmail on this one!

**Comment by mimugmail** [comment]
I get this when linking in Unbound:

`Mar 12 14:42:50 OPNsense opnsense: /services_unbound.php: The command '/usr/local/sbin/unbound -c '/var/unbound/unbound.conf'' returned exit code '1', the output was '/var/unbound/dnsbl.conf:1: error: syntax error read /var/unbound/unbound.conf failed: 1 errors in configuration file [1552398170] unbound[36366:0] fatal error: Could not read config file: /var/unbound/unbound.conf. Maybe try unbound -dd, it stays on the commandline to see more errors, or unbound-checkconf'
`

The file looks like this:

```
local-zone: "0-act.channel.facebook.com" redirect
local-data: "0-act.channel.facebook.com A 0.0.0.0"
local-zone: "0-edge-chat.facebook.com" redirect
local-data: "0-edge-chat.facebook.com A 0.0.0.0"
local-zone: "0.0.0.0" redirect
local-data: "0.0.0.0 A 0.0.0.0"
local-zone: "0.beta.facebook.com" redirect
local-data: "0.beta.facebook.com A 0.0.0.0"
local-zone: "0.r.msn.com" redirect
local-data: "0.r.msn.com A 0.0.0.0"
local-zone: "0.twitter.com" redirect
local-data: "0.twitter.com A 0.0.0.0"
local-zone: "000.0x1f4b0.com" redirect
local-data: "000.0x1f4b0.com A 0.0.0.0"
local-zone: "000.gaysexe.free.fr" redirect
local-data: "000.gaysexe.free.fr A 0.0.0.0"
local-zone: "000free.us" redirect
local-data: "000free.us A 0.0.0.0"
local-zone: "000owamail0.000webhostapp.com" redirect
local-data: "000owamail0.000webhostapp.com A 0.0.0.0"
local-zone: "000tristanprod.free.fr" redirect
local-data: "000tristanprod.free.fr A 0.0.0.0"
local-zone: "001.0x1f4b0.com" redirect
local-data: "001.0x1f4b0.com A 0.0.0.0"

```

**Comment by alectrocute** [comment]
@mimugmail That’s strange, let me look into this. The syntax looks fine to me, which leads me to believe there’s a character in there somewhere from the regex... hmm...

**Comment by alectrocute** [comment]
@mimugmail Haha, okay... derpy moment on my part. I think it’s missing a ‘server:’ on line 1. Try adding that and see if unbound starts without exception.

**Comment by beren12** [comment]
Why was this closed? Just wondering…

**Pull request closed without merging**